match/lib/match/spaceship_ensure.rb

```diff
@@ -74,13 +79,26 @@ def certificates_exists(username: nil, certificate_ids: [])
       UI.user_error!("To reset the certificates of your Apple account, you can use the `fastlane match nuke` feature, more information on https://docs.fastlane.tools/actions/match/")
     end
 
-    def profile_exists(type: nil, username: nil, uuid: nil, platform: nil)
-      # App Store Connect API does not allow filter of profile by platform or uuid (as of 2020-07-30)
-      # Need to fetch all profiles and search for uuid on client side
-      # But we can filter provisioning profiles based on their type (this, in general way faster than getting all profiles)
-      filter = { profileType: Match.profile_types(type).join(",") } if type
-      found = Spaceship::ConnectAPI::Profile.all(filter: filter).find do |profile|
-        profile.uuid == uuid
+    def profile_exists(profile_type: nil, name: nil, username: nil, uuid: nil, cached_profiles: nil)
```

Comment:
From a design perspective, I find it weird that we pass a list of cached profiles, find out it wasn't good enough, fetch them again, and our `cached_profiles` isn't updated.

Shouldn't we be instead be passing the `cache`around and getting rid of the `cached_profiles` parameters in several method calls?

So it makes me wonder if design wise, we should have organized the code differently for the cache to be more inbuilt and self-updating.

This is the biggest remark I have design wise on the PR. The rest are mostly nitpicks on small things there and there.

Reply:
Good one. Thanks for pointing out.

There is a logic in the runner `if spaceship && !spaceship.profile_exists(profile_type: profile_type`, which deletes the stored profile if it doesn't exist on the portal.
When using only the cache, there will be a situation when a new profile will not be listed in the cache, and the stored one will be deleted right after being created 😅. 

And `sigh` returns only a path to a stored profile, and currently, there is no way to get the Profile object from the portal.

I will introduce a new var `is_profile_created` to omit the `!spaceship.profile_exists` check because since the match created a profile, it exists 😅. This will allow to remove extra requests to the portal.

Reply:
Done in 39b784935b9129e43b9e19ca362d6d88198a9c1d.